core/services/head_tracker.go

```diff
@@ -104,103 +88,67 @@ func (ht *HeadTracker) logger() *logger.Logger {
 // subscribes to new heads, and if successful fires Connect on the
 // HeadTrackable argument.
 func (ht *HeadTracker) Start() error {
-	ht.headMutex.Lock()
-	defer ht.headMutex.Unlock()
-
-	if ht.started {
-		return nil
-	}
-
-	if err := ht.setHighestSeenHeadFromDB(); err != nil {
-		return err
-	}
-	if ht.highestSeenHead != nil {
-		ht.logger().Debugw(
-			fmt.Sprintf("Headtracker: Tracking logs from last block %v with hash %s", presenters.FriendlyBigInt(ht.highestSeenHead.ToInt()), ht.highestSeenHead.Hash.Hex()),
-			"blockNumber", ht.highestSeenHead.Number,
-			"blockHash", ht.highestSeenHead.Hash,
-		)
-	}
-
-	ht.subscriptionSucceeded = make(chan struct{})
+	return ht.StartOnce("HeadTracker", func() error {
```

Comment:
Does it still pass race detector without using the mutexes?

Reply:
the mutex was needed because it was accessing `ht.highestSeenHead` but i changed SetHighestSeenHeadFromDB now to instead return the same head.

But yes `go run -race core/main.go -- local n` also seems fine - can't see any warnings